### PR TITLE
Csv with data

### PR DESCRIPTION
**Pull request by jfsehuanes**
Created an independent module for creating .csv files. It is python2 as well as python3 suitable.

So far the .csv file contains the fundamental frequencies with their corresponding power in dB of all fishes detected in a recording.

### CODE REVIEW

File: thunderfish/csvmaker.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import numpy as np
+from .dataloader import load_data
+from .consistentfishes import consistent_fishes
+from .harmonicgroups import harmonic_groups
+from .powerspectrum import multi_resolution_psd
+
+
+def extract_main_freqs_and_db(data, samplerate):
```
Comment (janscience) [inline thunderfish/csvmaker.py]: You do not need this function. This is already done in thunderfish at line 224!

The only thing we might want to have is another function for harmonicgroups.py like fundamental_frequencies() that extracts fundamental frequencies and decibel power from the harmonic groups.

File: thunderfish/csvmaker.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    tmp = np.vstack([filtered_fishlist[e][0] for e in range(len(filtered_fishlist))])
+
+    freqs = tmp[:, 0]
+    pows = tmp[:, 1]
+    decibel = 10.0 * np.log10(pows)  # calculate decibel using 1 as P0
+    return freqs, decibel
+
+
+def freqs_and_db_csv(filename, freq, decibel):
+    header = ['fundamental frequency', 'dB']
```
Comment (janscience) [inline thunderfish/csvmaker.py]: How about a function

write_csv(filename, header, data)

that writes your data matrix as a cvs file?

**Review by janscience — COMMENTED** [review]
Some ideas...

File: thunderfish/csvmaker.py (hunk 1 above)
Comment (jfsehuanes) [reply thunderfish/csvmaker.py]: I agree, that I don't need it in thunderfish. But I do need it in the independent csvmaker.py module. So what do we suggest doing??

Comment (janscience) [reply thunderfish/csvmaker.py]: Why do you need it? For testing freqs_and_db_csv() you only need to supply a list of frequencies and powers. This function does not care about whether these lists have been computed by harmonic_groups() or supplied manually, like

```
freqs = [200, 350, 432]
pow = [0.87, 0.34, 7.2]
freqs_and_db_csv('test.csv', freqs, pow)
```

I would see the csvmaker as a module that only provides useful functions for writing out simple data (tables). It should not be used as as script, except for testing. And tests should be kept simple. In the future we might add functions for other file formats, like for example nix, or the .dat files from relacs. But that should not worry you for now.

File: thunderfish/thunderfish.py [hunk 3]
```diff
@@ -249,6 +250,11 @@ def thunderfish(audio_file, channel=0, output_folder='', verbosearg=0):
     output_plot(audio_file, pulse_fish_width, pulse_fish_psd, len(eod_times), median_IPI, inter_eod_intervals, raw_data,
                 samplerate, idx0, idx1, filtered_fishlist, period, time, mean_eod, std_eod, unit, psd_data, output_folder)
 
+    # Create csv file with main EODF and corresponding power in dB of detected fishes
+    freqs, db = extract_main_freqs_and_db(data, samplerate)
```
Comment (janscience) [inline thunderfish/thunderfish.py]: this line is not needed. A `fishlist` was already computed above.

File: thunderfish/thunderfish.py [hunk 4]
```diff
@@ -249,6 +250,11 @@ def thunderfish(audio_file, channel=0, output_folder='', verbosearg=0):
...
+    csv_name = os.path.splitext(os.path.basename(sys.argv[1]))[0] + '.csv'
```
Comment (janscience) [inline thunderfish/thunderfish.py]: We are going to write a couple of more csv files, therefore do

```
if wavefish:
    csv_name = os.path.splitext(os.path.basename(sys.argv[1]))[0] + '-eodfs.csv'
    freqs_and_db_csv(csv_name, freqs, db)
```

File: thunderfish/csvmaker.py (hunk 2 above)
Comment (janscience) [reply thunderfish/csvmaker.py]: Actually, I just checked out numpy's savetext function:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.savetxt.html#numpy.savetxt
It already can do everything we want. You may just write a really simple function that calls np. savetext with the appropriate parameters for writing a csv file.

**Review by janscience — COMMENTED** [review]
Suggestions for make csv file writing really samll and simple.

File: thunderfish/csvmaker.py (hunk 2 above)
Comment (jfsehuanes) [reply thunderfish/csvmaker.py]: Nice idea, Jan!! I think I came up with a fancy solution.

**Comment by jfsehuanes** [comment]
small question. Do the functions I wrote need Docu? I think I wrote them with names that seem pretty straight forward? Please advice if you disagree.

**Review by janscience — COMMENTED** [review]
every function needs documentation: you need to specify what you expect for the input arguments!

**Pull request merged**